node/block-procesing.go

```diff
@@ -0,0 +1,190 @@
+package node
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/harmony-one/harmony/consensus"
+	"github.com/harmony-one/harmony/core/types"
+	"github.com/harmony-one/harmony/shard"
+	"golang.org/x/sync/errgroup"
+)
+
+// postConsensusProcessing is called by consensus participants, after consensus is done, to:
+// 1. add the new block to blockchain
+// 2. [leader] send new block to the client
+// 3. [leader] send cross shard tx receipts to destination shard
+func (node *Node) postConsensusProcessing(
+	newBlock *types.Block, leader string,
+) error {
+
+	if node.Consensus.IsLeader() {
+
+		if err := node.Gossiper.AcceptedBlock(
+			node.Consensus.ShardID, newBlock,
+		); err != nil {
+			return err
+		}
+
+		// node.Gossiper.NewBeaconChainBlock(newBlock)
+
+		// if node.Consensus.ShardID == shard.BeaconChainShardID {
```

Comment:
why are these deleted?

Reply:
The topics ideas I changed up , the topics groups aren’t being used correctly , and currently it will cause deadlocks, it’s my next focus since getting foundation of sync working